src/commands/appSettings/downloadAppSettings.ts

```diff
@@ -6,27 +6,35 @@
 import { type StringDictionary } from "@azure/arm-appservice";
 import { confirmOverwriteSettings } from "@microsoft/vscode-azext-azureappservice";
 import { AppSettingsTreeItem, type IAppSettingsClient } from "@microsoft/vscode-azext-azureappsettings";
-import { AzExtFsExtra, type IActionContext } from "@microsoft/vscode-azext-utils";
+import { AzExtFsExtra, type IActionContext, type ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
 import * as vscode from 'vscode';
 import { functionFilter, localSettingsFileName } from "../../constants";
 import { viewOutput } from "../../constants-nls";
 import { ext } from "../../extensionVariables";
 import { getLocalSettingsJson, type ILocalSettingsJson } from "../../funcConfig/local.settings";
 import { localize } from "../../localize";
+import { isResolvedFunctionApp } from "../../tree/ResolvedFunctionAppResource";
 import type * as api from '../../vscode-azurefunctions.api';
+import { getEolWarningMessages } from "../createFunctionApp/stacks/getStackPicks";
 import { decryptLocalSettings } from "./localSettings/decryptLocalSettings";
 import { encryptLocalSettings } from "./localSettings/encryptLocalSettings";
 import { getLocalSettingsFile } from "./localSettings/getLocalSettingsFile";
 
-export async function downloadAppSettings(context: IActionContext, node?: AppSettingsTreeItem): Promise<void> {
+export async function downloadAppSettings(context: ISubscriptionActionContext, node?: AppSettingsTreeItem): Promise<void> {
     if (!node) {
         node = await ext.rgApi.pickAppResource<AppSettingsTreeItem>(context, {
             filter: functionFilter,
             expectedChildContextValue: new RegExp(AppSettingsTreeItem.contextValue)
         });
     }
 
+    const parent = node.parent;
     const client: IAppSettingsClient = await node.clientProvider.createClient(context);
+    if (isResolvedFunctionApp(parent)) {
```

Comment:
I think you can just make this a helper function, `showEolWarningIfNecessary` in the few places you call it.